### PR TITLE
docs: add unsubscribeevent to default list and add notes

### PR DESCRIPTION
**Pull request by danielpeintner**
The note is about best practices for unsubscribing

fixes https://github.com/w3c/wot-thing-description/issues/957

***
<a href="https://pr-preview.s3.amazonaws.com/danielpeintner/wot-thing-description/pull/960.html" title="Last updated on Oct 1, 2020, 8:36 AM UTC (5b7438e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/960/32ba69e...danielpeintner:5b7438e.html" title="Last updated on Oct 1, 2020, 8:36 AM UTC (5b7438e)">Diff</a>

### CODE REVIEW

**Comment by danielpeintner** [comment]
Note: If we agree on the way forward I will provide updates to index.template.html also.

File: index.html [hunk 1]
```diff
@@ -1218,7 +1218,9 @@ <h2>Core Vocabulary Definitions</h2>
 <code>unobserveproperty</code> or an  <a>Array</a> containing a combination of these terms.
     </span>
 </p>
-
+<p class="note">
+    It is considered to be good practice that each <code>observeproperty</code> has a corresponding <code>unobserveproperty</code> unless the protocol supports implicit unsubscription mechanisms (e.g., heartbeat to detect connection loss).
+</p>
```
Comment (relu91) [inline index.html:1223]: I'd also add this sentence: *Implementations might/should (?) use the "observerproperty" form as the default for "unobserverproperty" WoT interaction*. I think we can spare a little bit of redundancy here. For example:
```json
{
  "op": "observeproperty",
  "href": "mqtt://wot.example.com/level"
}
```
An implementation might infer that to *unobserveproperty* it needs to *unsubscribe* from `mqtt://wot.example.com/levelevent` topic.

**Review by relu91 — APPROVED** [review]
It looks good. I have only a single comment to open a discussion.

**Comment by zolkis** [comment]
Question: how can someone find the `unsubscribe` Form that matches its `subscribe` Form?

**Comment by danielpeintner** [comment]
> Question: how can someone find the `unsubscribe` Form that matches its `subscribe` Form?

It needs to be part of the same event/property (i.e. `"illuminance"` event)

```
    "events": {
        "illuminance": {
            "data":{"type": "integer"},
            "forms": [
                {
                    "href": "mqtt://192.168.1.187:1883/illuminance",
                    "contentType" : "text/plain",
                    "op" : "subscribeevent"
                },
                {
                    "href": "mqtt://192.168.1.187:1883/illuminanceUnsub",
                    "contentType" : "text/plain",
                    "op" : "unsubscribeevent"
                }
            ]
        }
    } 
```

Of course it is also possible that *one* form/href deals with both ops: `"op" : ["subscribeevent", "unsubscribeevent"]`  but it gets a bit trickier and you may need to know the current state...

**Comment by zolkis** [comment]
Of course they need to be part of the same interaction, but what if there are multiple protocols supported.
So we could take a look at the `subscribe` Form, and select an `unsubscribe` Form that has same scheme in `href`, or if multiple are found, then if there is one with also the same `host` and then `port`?

**Comment by egekorkan** [comment]
In a realistic scenario, the same protocol interface would handle the sub and unsub but TD allows that the protocols do it. I would not try to add a protocol equality check since if we think in detail:
- `subprotocol` must be same/usable
- `security` must be same/usable
- `href` must be _similar_, i.e. protocol, host (domain, local network, ...), port same/usable
- `contentType` same/usable, what if the implementation cannot parse XML that is needed only for unsubscription

There can be other points I miss as well

**Comment by zolkis** [comment]
We need a better algorithm for `unsubscribe` in Scripting than we have now, that is why I am asking.
I will factor out an algorithm for this, then it could be checked/tested/updated separately.

**Comment by sebastiankb** [comment]
from today's TD call:
- we call for more feedback on this discussion
- decide to merge in the next TD call then

**Comment by sebastiankb** [comment]
issue that is also related: #968

**Comment by sebastiankb** [comment]
from today's TD call:
- we decided to spend one more week for commenting
- there is a new issue from @zolkis #968 that can be common be discussed

**Review by sebastiankb — APPROVED** [review]

**Comment by sebastiankb** [comment]
From today's TD call:
- we will merge this PR
- however, the index.template.html has to be edited, @danielpeintner will work on this

**Comment by danielpeintner** [comment]
Note: this PR contains the necessary fixes in index.template.html and validation/td-validation.ttl and no update (expect aligning with master) to the other files like index.html.
Hence *someone* needs to run render script ... which contains many more changes due to other updates

**Comment by danielpeintner** [comment]
Close since a *fresh* PR https://github.com/w3c/wot-thing-description/pull/975 has been used instead

**Pull request closed without merging**